src/Frontend/components/app/widgets/LanguageSwitcher.js

```diff
@@ -10,40 +10,38 @@
 
 'use strict';
 
-import React, { Component } from 'react';
-import Translate from 'react-translate-component';
-import localeStore from '../../../store/localeStore';
-import NavDropdown from 'react-bootstrap/lib/NavDropdown';
-import LoadingDropDown from '../markup/LoadingDropDown';
-import DropDownItem from '../markup/DropDownItem';
-import { runAction } from 'sententiaregum-flux-container';
-import localeActions from '../../../actions/localeActions';
-import { GET_LOCALES, CHANGE_LOCALE } from '../../../constants/Locale';
-import { subscribeStores } from 'sententiaregum-flux-react';
-import Locale from '../../../util/http/Locale';
+import React, { Component, PropTypes  } from 'react';
+import Translate                        from 'react-translate-component';
+import NavDropdown                      from 'react-bootstrap/lib/NavDropdown';
+import LoadingDropDown                  from '../markup/LoadingDropDown';
+import DropDownItem                     from '../markup/DropDownItem';
+import Locale                           from '../../../util/http/Locale';
 
 /**
  * Widget which changes the user locale.
  *
  * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
  */
-class LanguageSwitcher extends Component {
-  /**
-   * Lifecycle hook triggers the action to gather locale information.
-   *
-   * @returns {void}
-   */
-  componentDidMount() {
-    runAction(GET_LOCALES, localeActions, []);
-  }
+export default class LanguageSwitcher extends Component {
+
+  static PropTypes = {
+    actions: PropTypes.object.isRequired
+  };
+
+  static contextTypes = {
+    store: React.PropTypes.object
```

Comment:
I think that it's better to inject the locale state directly rather than passing the entire store down the component tree.
How about creating a `LocaleContainer` component which contains all the `connect()` and redux-related stuff and is rendered in the `MenuWrapper`? This `LocaleContainer` could render the `LanguageSwitcher`, so no need to pass down the locale actions through the MenuWrapper and fetch the store directly in this component.